src/supernova/utils.rs

```diff
@@ -117,4 +137,32 @@ mod test {
       }
     }
   }
+
+  #[test]
+  fn test_get_selector() {
+    for n in 1..4 {
+      for selected in 0..n {
+        let mut cs = TestConstraintSystem::<Base>::new();
+
+        let allocated_target =
+          alloc_const(&mut cs.namespace(|| "target"), Base::from(selected as u64)).unwrap();
+
+        let selector_vec = get_selector_vec_from_index(&mut cs, &allocated_target, n).unwrap();
```

Comment:
I don't think this will ever allocate a target that is out of range, so this test is misleading. The condition `selected < n` will always be truce, since selected is always taken from `0..n`.

Reply:
I think you're referring to the test below. I'm extending the range of `selected` to `2*n` to catch the unsatisfiable cases. 
I've also changed the `is_valid` condition to only run in the satisfying case, and checking that the correct bit is set.